src/main/java/org/spongepowered/api/schematic/SchematicEntity.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.schematic;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.entity.EntityType;
+
+public interface SchematicEntity extends DataHolder, DataSerializable {
```

Comment:
Rotations is (according to Deamon) stored in the custom data. The Vector3d is already relative to the origin (which is basically the offset + lower bound).
> On Apr 4, 2016, at 4:39 PM, Gabriel Harris-Rouquette <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/schematic/SchematicEntity.java <https://github.com/SpongePowered/SpongeAPI/pull/1172#discussion_r58453840>:
> 
> > + * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> > + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> > + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> > + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> > + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > + * THE SOFTWARE.
> > + */
> > +package org.spongepowered.api.schematic;
> > +
> > +import com.flowpowered.math.vector.Vector3d;
> > +import org.spongepowered.api.data.DataHolder;
> > +import org.spongepowered.api.data.DataSerializable;
> > +import org.spongepowered.api.data.persistence.DataBuilder;
> > +import org.spongepowered.api.entity.EntityType;
> > +
> > +public interface SchematicEntity extends DataHolder, DataSerializable {
> What SchematicEntity needs is a simple Vector3d relative to the offset perhaps, and of course rotations.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/SpongePowered/SpongeAPI/pull/1172/files/48fdb2e4ac20bb17b352baf78dbcafca57386ea3#r58453840>